### ISSUE TITLE
Bioconda installation

### ISSUE THREAD

**Issue by ChemaMD** [vanheeringen-lab/gimmemotifs#271]
Hi,

Thanks for the great package. I'm trying to install GimmeMotifs via conda and it keeps failing (in different computers and users). The command I run, as per guides in readthedocs:

`conda create -n gimme gimmemotifs`

Output:

`Collecting package metadata (current_repodata.json): done`
`Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.`
`Collecting package metadata (repodata.json): done`
`Solving environment: /`

In case it is useful, below the info of my conda installation.

Thanks for your help!

Chema

====== `conda info` =====

`          conda version : 4.13.0`
`    conda-build version : 3.21.9`
`         python version : 3.8.13.final.0`
`       virtual packages : __osx=10.16=0`
`                        __unix=0=0`
`                          __archspec=1=x86_64`
`       base environment : /opt/anaconda3  (writable)`
`      conda av data dir : /opt/anaconda3/etc/conda`
`  conda av metadata url : None`
`           channel URLs : https://conda.anaconda.org/conda-forge/osx-64`
`                          https://conda.anaconda.org/conda-forge/noarch`
`                          https://conda.anaconda.org/bioconda/osx-64`
`                         https://conda.anaconda.org/bioconda/noarch`
`                          https://repo.anaconda.com/pkgs/main/osx-64`
`                          https://repo.anaconda.com/pkgs/main/noarch`
`                          https://repo.anaconda.com/pkgs/r/osx-64`
`                          https://repo.anaconda.com/pkgs/r/noarch`
`               platform : osx-64`
`             user-agent : conda/4.13.0 requests/2.28.0 CPython/3.8.13 Darwin/21.4.0 OSX/10.16`
`               UID:GID : 502:20`
`             netrc file : None`
`           offline mode : False`

**Comment by paolabc** [comment]
Yes, I had the same problem. Maybe, a solution is to install it via source code. Download it from here https://anaconda.org/bioconda/gimmemotifs/files and install it by using  the command conda install offline name_of_the_file_package. There are other alternatives to install this tool and it is described in the instalation section here https://gimmemotifs.readthedocs.io/en/stable/installation.html. :)

**Comment by paolabc** [comment]
I also would like to know why this problem of conda installation for this tool is happened. Conda install is an easy and friendly way.

**Comment by siebrenf** [comment]
Hey @ChemaMD and @paolabc,
GimmeMotifs if using a big bunch of big packages for motif discovery. It seems Conda is really unhappy with one or both of those two. I'm trying to fix the root cause (by splitting the gimmemotifs conda recipe), but this is still a work in progress...

Aside from @paolabc's excellent suggestions, here are two additional methods to fix Conda (for all big packages, not just gimmemotifs):
- Conda itself is working on these kinds of issues, and has an experimental fix that you can find [here](https://github.com/conda-incubator/conda-libmamba-solver). 
- You can use conda to install mamba, and then use mamba as if you're using conda (but faster!):
  ```
  conda install -c conda-forge conda=4.13 mamba=0.24
  mamba create -n gimme gimmemotifs
  ```

**Comment by paolabc** [comment]
Hi
Thank you for your quicly reply. I got it. And I just tried it here and it works as well. :)

**Comment by ChemaMD** [comment]
Hi, thanks for the suggestions, I'll def give it a go!

**Comment by shaunmahony** [comment]
I found the same problem with conda. The pip install method works, however.

**Comment by IanCodes** [comment]
Has there been any movement on this issue?  
I tried installing via mamba, but get an out of memory error.  
I also tried the local install method by downloading files from Anaconda:
https://anaconda.org/bioconda/gimmemotifs/0.17.1/download/linux-64/gimmemotifs-0.17.1-py38h8ded8fe_1.tar.bz2

`conda install --offline gimmemotifs-0.17.1-py38h8ded8fe_1.tar.bz2`

This appeared to work:
`Preparing transaction: done`  
`Verifying transaction: done`  
`Executing transaction: done`

However, trying to run gimme results in:
`(gimme) $ gimme`
`Traceback (most recent call last):`
`  File "/mnt/fls01-home01/mqbssid3/.conda/envs/gimme/bin/gimme", line 8, in <module>`
`    from gimmemotifs.cli import cli`
`ModuleNotFoundError: No module named 'gimmemotifs'`

I would like to revisit this tool, but conda isn't helping.

Thanks.

**Comment by IanCodes** [comment]
> 
Thank you @siebrenf.  I tried the mamba method again on a different system.  It installed, but I get the following error:
`ImportError: libcrypto.so.1.0.0: cannot open shared object file: No such file or directory`

Any thoughts?

I am using Ubuntu 20 if that helps.

Thank you.

**Comment by siebrenf** [comment]
Some of our dependencies are outdated on bioconda, and throw this error occasionally (it is likely one of the ucsc tools). This behavior is not consistent, so you might be OK if you run it again...

**Comment by IanCodes** [comment]
I thought I made progress, but get the following:
`
Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.31=0
  - feature:|@/linux-64::__glibc==2.31=0
  - gimmemotifs -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.31
`

I am not sure how to interpret this, but don't I have a never version of glibc than requested?

Thank you.

**Comment by siebrenf** [comment]
That's a conda error right? It seems you are trying to install something new in the environment that is clashing with something already in there... Easiest solution is to simply create a new environment :)

Btw, i am trying to release a new version of gimmemotifs on bioconda that is easier to install (0.17.2). Hopefully done on monday!

**Comment by IanCodes** [comment]
@siebrenf thanks for the reply.  It was a new environment:
`conda create -n gimme python=3`

I'll try the new release next week.

**Comment by IanCodes** [comment]
Unfortunately, the new version doesn't install either.
`conda create -n gimme python=3`
`conda install gimmemotifs` 

 Resulted in "Solving environment:" running for three days before I stopped it.

**Comment by siebrenf** [comment]
Hey Ian, 

This command should work within minutes: `mamba create -n gimme gimmemotifs=0.17.2`. 
You can specify which python version you want with it (3.7-3.10, default is 3.10. I tested with 3.8):  `mamba create -n gimme gimmemotifs=0.17.2 python=3.8`.

**Comment by IanCodes** [comment]
@siebrenf just to say I successfully installed gimmemotifs via mamba.  Thank you for your help.

**Comment by foryoubar** [comment]
Hello，I’ve tried to install gimmemotifs according to the instructions on the official website, I ran the following command：
$ conda config --add channels defaults
$ conda config --add channels bioconda
$ conda config --add channels conda-forge
$ conda install -c conda-forge "conda>=4.12" "mamba>=0.27"
After mamba installed, I ran "mamba install gimmemotif", but got a " Could not solve for environment specs
The following package could not be installed
└─ gimmemotif   does not exist (perhaps a typo or a missing channel)"
Could you please help me deal with this bug? Thank you!

**Comment by simonvh** [comment]
Can you try:

mamba install gimmemotifs

(With an s at the end)

**Comment by foryoubar** [comment]
Sure. I tried this command ”mamba install gimmemotifs“before, and it showed the same error.

**Comment by siebrenf** [comment]
Your current conda environment might be incompatible. You can create a fresh environment with 

`mamba create -n gimme gimmemotifs`

**Comment by foryoubar** [comment]
Thank you sincerely for your patient reply!&nbsp;I will follow your suggestion and try to install it again later.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "vanheeringen-lab/gimmemotifs"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月11日(星期二) 下午3:15
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [vanheeringen-lab/gimmemotifs] Bioconda installation (Issue #271)





 
Your current conda environment might be incompatible. You can create a fresh environment with
 
mamba create -n gimme gimmemotifs
